python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,730 @@
+            List of strings describing the edge features.
+        """
+        return self._etype_features[etype]
+
+    def get_etype_feature_files(self, etype, feature_name):
+        """Retrieves a tuple, with the metadata for edge feature.
+
+        Argument
+        --------
+        etype: str
+            Edge type from the input graph.
+        feature_name: str
+            Edge feature name associated with the ``etype`` edge type.
+
+        Returns
+        -------
+        tuple: (str, str, list(str))
+            File type (valid values are ``numpy``or ``parquet``), delimiter
+            (a valid delimiter) and list of paths for filenames in which
+            edge features are stored.
+        """
+        return self._etype_feature_files[(etype, feature_name)]
+
+    def get_ntype_features(self, ntype):
+        """Retrieves a list of node features associated with ``ntype``.
+
+        Argument
+        --------
+        ntype: str
+            Node type from the input graph.
+
+        Returns
+        ------
+        list(str): list of strings
+            List of strings describing the node features.
+        """
+        return self._ntype_features[ntype]
+
+    def get_ntype_feature_files(self, ntype, feature_name):
+        """Retrieves a tuple, with the metadata for node feature.
+
+        Argument
+        --------
+        ntype: str
+            Node type from the input graph.
+        feature_name: str
+            Node feature name associated with the ``ntype`` node type.
+
+        Returns
+        -------
+        tuple: (str, str, list(str))
+            File type (valid values are ``numpy``or ``parquet``), delimiter
+            (a valid delimiter) and list of paths for filenames in which
+            node features are stored.
+        """
+        return self._ntype_feature_files[(ntype, feature_name)]
+
+    ############################## Properties #################################
+
+    @property
+    def data(self):
+        """Property to get the schema object read from the input metadata
+        file.
+        """
+        return self._data
+
+    @property
+    def id_etype_map(self):
+        """Property to get the dictionary which stores (edge_id, etype) as
+        the key-value pairs.
+        """
+        return self._id_etype_map
+
+    @property
+    def id_ntype_map(self):
+        """Property to get the node-id <-> node type map."""
+        return self._id_ntype_map
+
+    @property
+    def etypes(self):
+        """Property to get the list of edges types of the input graph."""
+        return self._etypes
+
+    @property
+    def etype_id_map(self):
+        """Property to get the dictionary which stores (etype, edge_id) as
+        the key-value pairs.
+        """
+        return self._etype_id_map
+
+    @property
+    def etype_features(self):
+        """Property to get a list of feature names for each edge type."""
+        return self._etype_features
```

Comment:
why do you define both properties and methods for this?

Reply:
Please note that property returns the entire data structure, where as the get method will lookup into the dictionary and returns value associated with that key. Its for ease of use for the application.

Reply:
i don't see how it is easier.